### ISSUE TITLE
NaN Losses for GradientDescentOptimizer and MomentumOptimizer

### ISSUE THREAD

**Issue by BelhalK** [tensorflow/probability#833]
In cifar10_bnn.py,:
For fake data, using tf.compat.v1.train.GradientDescentOptimizer and tf.compat.v1.train.MomentumOptimizer quickly leads to NaN losses and KL.

It works for very small learning rates but then VERY slowly converge.

Do you have an idea why? 
I have to put a learning rate of 10^-9 to make it work.  (And it is very sensitive)

Thanks!
Belhal
